ItunesSearch/ItunesSearch/SearchResultController.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  SearchResultController.swift
+//  ItunesSearch
+//
+//  Created by Keri Levesque on 2/11/20.
+//  Copyright © 2020 Keri Levesque. All rights reserved.
+//
+
+import Foundation
+
+class SearchResultController {
+    let baseURL = URL(string: "https://itunes.apple.com/search")! // base URL for the itunes search API
+    
+    var searchResults: [SearchResult] = [] // data source for the table view
+
+    enum HTTPMethod: String {
+        case get = "GET"
+        case put = "PUT"
+        case post = "POST"
+        case delete = "DELETE"
+    }
+    func performSearch(searchTerm: String, resultType: ResultType, completion: @escaping (_ error: Error?) -> Void) {
+        // create a URL components objects from our base URL
+        var urlComponents = URLComponents(url: baseURL, resolvingAgainstBaseURL: true)
+        
+        let searchTermQueryItem = URLQueryItem(name: "term", value: searchTerm)
+
+        let resultTermQueryItem = URLQueryItem(name: "type", value: resultType.rawValue)
```

Comment:
In the API there is not "type" as a Parameter Key. The one you want is "entity". So replace "type" with "entity"